stories/RefreshCache.stories.js

```diff
@@ -128,3 +129,74 @@ class AppWithRefresh extends Component {
 }
 
 stories.add('with a refresh button', () => <AppWithRefresh />);
+
+class StopWatch extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { count: 0 };
+  }
+
+  componentWillReceiveProps = nextProps => {
+    if (nextProps.refresh) {
+      this.setState({ count: 0 });
+    }
+  };
+
+  componentDidMount = () => {
+    setInterval(
+      () =>
+        this.setState(prevState => ({
+          count: prevState.count + 1,
+        })),
+      1000
+    );
```

Comment:
We don't need to set two intervals, only one should be sufficient. Set an interval of 1000 and reset it when it reach the desired value. Then you can get rid of this component I think. Try to also use the same reset strategy as explained above.